include/dice/template-library/flex_array.hpp

```diff
@@ -159,15 +272,19 @@ namespace dice::template_library {
 		constexpr flex_array(Iter first, Sent last) {
 			size_t ix = 0;
 			while (first != last) {
-				if (ix >= max_size()) [[unlikely]] {
-					throw std::length_error{"flex_array::flex_array: maximum size exceeded"};
+				if constexpr (has_max_extent) {
+					if (ix >= max_size()) [[unlikely]] {
```

Comment:
Would be better to check if this can happen before the loop.

Reply:
Doing it before the loop now, iff iterator is random access